client/src/components/CreateNewPost.js

```diff
@@ -0,0 +1,54 @@
+import React from 'react'
+
+class CreateNewPost extends React.Component {
+    constructor(props){
```

Comment:
You don't have to do this unless you are using props inside of the constructor
```suggestion
    constructor(){
```